backend/packages/Upgrade/src/api/controllers/validators/MarkExperimentValidator.v4.ts

```diff
@@ -1,21 +1,135 @@
-import { IsNotEmpty, IsDefined } from 'class-validator';
-import { MARKED_DECISION_POINT_STATUS, PAYLOAD_TYPE } from 'upgrade_types';
+import { Type } from 'class-transformer';
+import { IsNotEmpty, IsDefined, IsString, IsOptional, IsEnum, IsObject, ValidateIf, ValidateNested, ValidationOptions, registerDecorator } from 'class-validator';
+import { EXPERIMENT_TYPE, MARKED_DECISION_POINT_STATUS, PAYLOAD_TYPE } from 'upgrade_types';
+
+export const IsAssignedFactorRecord = (validationOptions?: ValidationOptions) => {
```

Comment:
You can remove export if this decorator is not used outside this file.

Reply:
Code updated